src/concrete/ml/sklearn/base.py

```diff
@@ -1328,6 +1339,266 @@ def __init__(self, n_bits: Union[int, Dict[str, int]]):
 
         BaseEstimator.__init__(self)
 
+    # pylint: disable=too-many-locals,too-many-statements,too-many-branches
+    @classmethod
+    def from_sklearn_model(
+        cls,
+        sklearn_model: sklearn.base.BaseEstimator,
+        X: Optional[numpy.ndarray] = None,
+        n_bits: int = 8,
+    ):
+        """Build a FHE-compliant model using a fitted scikit-learn model.
```

Comment:
I would try to divide the following function into smaller functions if possible, as it's not always easy to follow what's going on !